### PR TITLE
Add the template title and type to the site hub

### PR DESCRIPTION
**Pull request by youknowriad**
Builds on top of #46700 
Related #36667

## What?

This PR adds the template title and type to the site hub in the site editor to match the proposed designs.

## Why?

These titles and type gives clarity about the entity that is currently being shown on the site editor frame.

## How?

 - I've extracted a `useEditedEntityRecord` hook from the DocumentActions component and used it in the site hub to grab the title.
 - I've extracted the `SiteHub` component from the `Layout` component as it was getting a bit complex.

## Testing Instructions

 1- Open the site editor
 2- Navigate templates, template parts...

Screenshot 2022-12-22 at 11 34 53 AM

### CODE REVIEW

File: packages/edit-site/src/components/header-edit-mode/document-actions/index.js [hunk 1]
```diff
@@ -126,14 +102,19 @@ export default function DocumentActions() {
...
 			<div className="edit-site-document-actions">
 				{ __( 'Template not found' ) }
 			</div>
 		);
 	}
 
+	const entityLabel =
+		record.type === 'wp_template_part'
+			? __( 'template part' )
+			: __( 'template' );
```
Comment (youknowriad) [inline packages/edit-site/src/components/header-edit-mode/document-actions/index.js:116]: There was a bug in trunk where these strings were not translated.

**Comment by jasmussen** [comment]
Nice, this is feeling solid to me. One small issue, the text remains in-editor:

status

I think there's an open question on whether the type or the name should be on top or below, and white or gray. I know @jameskoster has a lot of thoughts on this, and he's AFK for the moment. But I also think we can arguably launch with one set, and then revisit. Similarly, there are some margins and site logo dimensions we can polish, nothing that I think need block this. 

One thing that I expect we'll want to refine is eliding of text, and how that works for templates with very long titles. I think we can use `text-overflow: ellipsis;` for those and it will be fine. But I also think we can bump the width of the left navigation from 320px (current) to 360px. There isn't a huge value in a massively giant browse-view window, and once you click "edit", you get a full-screen regardless. So I think the extra space is worth it. We can always refine details in responsive contexts.

**Comment by youknowriad** [comment]
Fixed the bug and added the text ellipsis.

**Comment by jasmussen** [comment]
Seems good to me!

File: packages/edit-site/src/components/header-edit-mode/document-actions/index.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/edit-site/src/components/header-edit-mode/document-actions/index.js:116]: Couldn't we rely o the post type labels provided by the rest API?

**Review by jorgefilipecosta — COMMENTED** [review]
In a small window size, things look good:
image
In big window sizes too.
But for medium window sizes, there are some ranges where things look strange, and the site preview is not seen:
image

File: packages/edit-site/src/components/header-edit-mode/document-actions/index.js (hunk 1 above)
Comment (youknowriad) [reply packages/edit-site/src/components/header-edit-mode/document-actions/index.js:116]: These labels are only available in plural form for all users. If we can reword here to use plural that would work, but let's leave this separate and just keep the bug fix for now.

**Comment by youknowriad** [comment]
Good catch @jorgefilipecosta pushed a fix for the breakpoint issue.

**Review by jasmussen — APPROVED** [review]
This one is important for #46700. The colors and sequence of title + type we can refine, but this works well as is, with eliding and such.

I'd still like to see a 360px wide sidebar. There's room for it.

**Review by jorgefilipecosta — APPROVED** [review]
Works well and the breakpoint issue is fixed 👍

**Pull request merged**